src/main/java/org/variantsync/diffdetective/diff/result/CommitDiffResult.java

```diff
@@ -24,4 +32,12 @@ public static CommitDiffResult Failure(DiffError error, String message) {
                 List.of(DiffResult.Failure(error, message).unwrap().getFailure())
         );
     }
+
+    public static CommitDiffResult fromCommitInRepository(final String commitHash, final Repository repository) throws IOException {
```

Comment:
Maybe it's more coherent to place this method into `GitDiffer` because there are already many such utility functions in this class.

Reply:
Yes that would definitely be another place.

Reply:
Moved and renamed to `createCommitDiff` as there is already a method which does the same except for accepting a `RevCommit`.